lib/splitclient-rb/engine/matchers/between_semver_matcher.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module SplitIoClient
+  class BetweenSemverMatcher < Matcher
+    MATCHER_TYPE = 'BETWEEN_SEMVER'
+
+    attr_reader :attribute
+
+    def initialize(attribute, start_value, end_value, logger, validator)
+      super(logger)
+      @validator = validator
+      @attribute = attribute
+      @semver_start = SplitIoClient::Semver.build(start_value, logger)
+      @semver_end = SplitIoClient::Semver.build(end_value, logger)
+      @logger = logger
+    end
+
+    def match?(args)
+      return false unless verify_semver_arg?(args, 'BetweenSemverMatcher')
+
+      value_to_match = SplitIoClient::Semver.build(args[:attributes][@attribute.to_sym], @logger)
+      unless !value_to_match.nil? && !@semver_start.nil? && !@semver_end.nil?
```

Comment:
why are you negating? you can do this `if value_to_match.nil? || @semver_start.nil? || !@semver_end.nil?`.
If at least one is nil it simply returns `false`